clang/lib/DPCT/AnalysisInfo.cpp

```diff
@@ -2728,7 +2728,10 @@ void CallFunctionExpr::buildInfo() {
 
   if (DpctGlobalInfo::isOptimizeMigration() && !FuncInfo->isInlined() &&
       !FuncInfo->IsSyclExternMacroNeeded()) {
-    FuncInfo->setAlwaysInlineDevFunc();
+    if (FuncInfo->isKernel())
+      FuncInfo->setForceInlineDevFunc();
+    else
+      FuncInfo->setAlwaysInlineDevFunc();
```

Comment:
What difference between setForceInlineDevFunc() and setAlwaysInlineDevFunc()

Reply:
setForceInlineDevFunc will only add the force inline for the top global kernel.
And setAlwaysInlineDevFunc will add inline for other device functions if available.